### PR TITLE
kicad: init unstable at version 2017-12-11

### PR DESCRIPTION
**Pull request by berce**
###### Motivation for this change

Files are not forward compatible, so when cooperating with someone who needs features of the Nightly Development Build, you also need this newer version.
In contrast with the current package for the stable release of KiCad, this package also enables oce, ngspice and scripting support.
This package does not provide the component libraries (schematics, PCB footprints, 3D models). They can be obtained from [KiCad](http://kicad-pcb.org/libraries/download/), unpacked and referenced at runtime.

It is called unstable and not nightly because it doesn't automatically track upstream nightly development builds.

wxPython headers are needed at buildtime. This PR includes  a necessary change to wxPython. That change is also in a separate PR: #32647

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by berce** [comment]
I took the liberty of adding @viric as a maintainer. Please approve :-)

**Comment by berce** [comment]
It failed because I tried to set the version of wxGTK used by wxPython in a wrong way.
For testing I had set `wxGTK = wxGTK30` in all-packages.nix. That worked.
How can I specify the version of a dependency of wxPython in `inherit (pythonPackages) wxPython;`

File: pkgs/applications/science/electronics/kicad/unstable.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{ config, wxGTK, lib, pkgs, stdenv, fetchFromGitHub, cmake, mesa, zlib
```
Comment (FRidh) [inline pkgs/applications/science/electronics/kicad/unstable.nix]: - `config` is not used
- `pkgs` should be removed

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -19017,6 +19017,12 @@ with pkgs;
     boost = boost160;
   };
 
+  kicad-unstable = callPackage ../applications/science/electronics/kicad/unstable.nix {
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: `python.pkgs.callPackage`

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -19017,6 +19017,12 @@ with pkgs;
...
+    wxGTK = wxGTK30;
+    boost = boost160;
+    inherit (pythonPackages) wxPython;
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: so this can be removed

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (berce) [reply pkgs/top-level/all-packages.nix]: KiCad optionally uses python for scripting. It is not a python application.
Is this change still required?
Also no other package in `top-level.nix` uses the proposed code.

Comment (FRidh) [reply pkgs/top-level/all-packages.nix]: there are a few that use `pythonPackages.callPackage` which is the same.

Yes, this should be used. Top-level packages should not have any `inherit (pythonPackages) foo;` as that risks mixing Python 2 and 3 packages (which was a common issue 3+years ago).

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (berce) [reply pkgs/top-level/all-packages.nix]: Leaving out the `wxGTK = wxGTK30` line makes it default to wxGTK28. Even when using `python.pkgs.callPackage`. Then compiling fails due to finding wxwidgets 2.8.12 and at least 3.0.0 is required.

Comment (FRidh) [reply pkgs/top-level/all-packages.nix]: I was referring only to 
```
inherit (pythonPackages) wxPython;
```

File: pkgs/applications/science/electronics/kicad/unstable.nix (hunk 1 above)
Comment (berce) [reply pkgs/applications/science/electronics/kicad/unstable.nix]: Done locally.
Should I squash it in the previous commit to update the PR, or add a new commit?

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (berce) [reply pkgs/top-level/all-packages.nix]: If you meant removing just the last line (`inherit...`), it seems to compile well.
wxPython related build problems only appear at 75% of compiling, so it 'll take some time to have full confirmation.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (berce) [reply pkgs/top-level/all-packages.nix]: Then I 'll use `pythonPackages.callPackage`for uniformity. Unless you object.
If you object, I propose to also change the others in a separate PR.

**Comment by berce** [comment]
All proposed changes have been applied.

**Pull request merged**